### ISSUE TITLE
Android Firebase analytics don't show up

### ISSUE THREAD

**Issue by jonathanhollander** [distriqt/ANE-Firebase#122]
I realize you have another issue with pretty much the exact same title, but I think my issue may be different.

I also didn't want to flood the other thread with my "story" explaining what's happening.

I'm trying to get Analytics and Crashlytics working on Android but having difficulties. 

To best explain my current scenario, I need to inform you of everything leading up to the current state.

I first started out, just trying to get Push notifications working with Firebase on Android. At this time I had a bundle id of "com.laaser.incidentviewer" (even though for AIR i add the "air." prefix) . I setup everything correctly in firebase and in the client, and successfully got push notifications to work. When I would login to firebase and see the overview, I would see the "1 app in project" and see my "air.com.laaser.incidentviewer" Android app and I could see Analytics working with "Daily Active users", "monthly users", etc. It was all working great.

Eventually I was tasked to make this work on iOS, but at the same time I got another requirement to change the bundle ID to "com.laaser.psapliv".

SO, I went into firebase and created 2 additional apps (because they are tied to the bundle ID), 1 for Android, 1 for iOS.... so that left me with 3 apps listed (including the original).

I then went through the configuration process for iOS (just adding the GoogleService-info.plist) and then redid the confirguration for Android (compiling the CustomResources ANE with the details from the Google-services.json, just like I did the first time).

I eventually got push notifications working for both Android and iOS on this new bundle id "com.laaser.psapliv", but I noticed that it seemed like the "Analytics" part for Android of the new bundle ID didn't seem to be coming through. I waited 2 more days, still nothing. 

I could see analytics coming through for the original Android "com.laaser.incidentviewer" and the new iOS "com.laaser.psapliv", but I didnt see anything ever for the new Android "com.laaser.psapliv".

Fast Forward a week, I still don't see anything coming through for Analytics on the Android "com.laaser.psapliv"... but just to confirm, I am 100% able to get push notifications on the new Android "com.laaser.psapliv", so that's why I'm confused why it isn't working.

I have since then deleted the old Android "com.laaser.incidentviewer" because we no longer need it, so now I have 2 apps listed, 1 Android, 1 iOS, but no Analytics for the Android version. 

Here is what I see when I login to firebase:
![emptyfirebase](https://user-images.githubusercontent.com/7573833/33634390-f8df3f66-d9e1-11e7-97f4-c71003803b38.png)

I'm at a lost for what to do. I've verified the info in the "google-services.json" is correctly copied into the "values.xml" (because also wouldn't the push notifications not be working if it was wrong)?

So long story short, I originally got Analytics working on my Android app, but when I added a new app with a new bundle ID, I could never get the Analytics to work again, even though I'm getting Firebase push notifications on the new bundle ID.

I'm attaching the google-services.json and the values.xml file (as zips) that I used to compile the Custom Resources ANE.
[google-services.zip](https://github.com/distriqt/ANE-Firebase/files/1533194/google-services.zip)

[values.zip](https://github.com/distriqt/ANE-Firebase/files/1533195/values.zip)

**Comment by marchbold** [comment]
With your change of id I'm assuming you are using the AIR_NOANDROIDFLAIR flag? The values in the google services you sent still have the air prefix though? Are you sure you are using the correct values?
 
You can investigate a little using `adb` and the Debug View in the firebase console.

Firstly make sure you've [enabled developer mode on your Android device](https://airnativeextensions.github.io/tutorials/device-debugging)

Open up a terminal/console and use adb to enable debug analytic events:

```
adb shell setprop debug.firebase.analytics.app air.com.laaser.psapliv
``` 

This will enable debugging mode and will send events almost immediately to the Debug View on the console:

![screen shot 2017-12-06 at 10 09 55](https://user-images.githubusercontent.com/442356/33637684-321e6a90-da6e-11e7-9900-6fc95b117053.png)

Additionally you can check the logs to see if there are any errors. It helps to set logging to verbose:

```
adb shell setprop log.tag.FA VERBOSE
```

Then restart your application and watch the logs:

```
adb logcat
```

You should see something like:

```
12-06 10:16:45.251 28709 28709 I FA      : App measurement is starting up, version: 11020
12-06 10:16:45.251 28709 28709 I FA      : To enable debug logging run: adb shell setprop log.tag.FA VERBOSE
12-06 10:16:45.258 28709 28709 V FA      : Collection enabled
12-06 10:16:45.259 28709 28709 V FA      : App package, google app id: air.com.distriqt.test, 1:XXXXXXXXXXXXX:android:XXXXXXXXXXXXXX
12-06 10:16:45.259 28709 28709 I FA      : Faster debug mode event logging enabled. To disable, run:
12-06 10:16:45.259 28709 28709 I FA      :   adb shell setprop debug.firebase.analytics.app .none.
12-06 10:16:45.259 28709 28709 D FA      : Debug-level message logging enabled
12-06 10:16:45.264 28709 28709 V FA      : Registered activity lifecycle callback
```

Check the app id is correct and watch for any errors when logging events.

**Comment by jonathanhollander** [comment]
sorry I think I have confused you. We never dropped the air prefix, so technically i guess we went from "air.com.laaser.incidentviewer" to "air.com.laaser.psapliv". also I'm not sure what you mean by the AIR_NOANDROIDFLAIR flag. Thanks for showing how to do the debugging, I couldn't figure out that today when I saw Firebase talking about those debug events that should show instantly. I will do what you mentioned and let you know how it goes. Thanks again

**Comment by marchbold** [comment]
Ah okay, yes I misread the app id change. (Setting AIR_NOANDROIDFLAIR drops the `air.` prefix from your package name if you ever require that).

Try out the debugging tools, and check that the id that's displayed matches. 

The other issue could be with the signature of the certificate you are using to sign your application, double check you've added the correct one to the Firebase console.

**Comment by jonathanhollander** [comment]
I never knew I needed to add a certificate for Android to the firebase console. I don't see that in your instructions either. Where do I do that? I only see the options for adding certificates on iOS.

**Comment by jonathanhollander** [comment]
BTW... I recently saw I could do a force crash by running FirebaseCrash.service.forceCrash();

The first time I went to try it, i saw a message saying I needed to first run "FirebaseCrash.init(key)"... I never did this "init" for FirebaseCrash nor Firebase... I didn't see it in the instructions? Although it is working for iOS, do i need to do that here?

**Comment by jonathanhollander** [comment]
I tried doing some logging events and I see this:


12-06 12:07:08.897 2795-2856/? D/InputReader: Input event(1): value=1 when=1882020527699000
12-06 12:07:08.897 2795-2856/? I/InputReader: Touch event's action is 0x0 (deviceType=0) [pCnt=1, s=0.16227 ] when=1882020527699000
12-06 12:07:08.897 2795-2856/? D/InputReader: lastThreadEndTime = 1882006007967300, currentThreadStartTime = 1882006007978684, diff = 0
12-06 12:07:08.897 2795-2855/? I/InputDispatcher: Delivering touch to (26417): action: 0x0, toolType: 1
12-06 12:07:08.897 26417-26417/? D/ViewRootImpl: ViewPostImeInputStage ACTION_DOWN
12-06 12:07:08.897 2795-2914/? E/lights: write_int failed to open -1
12-06 12:07:08.997 2795-2856/? D/InputReader: Input event(1): value=0 when=1882020620369000
12-06 12:07:08.997 2795-2856/? I/InputReader: Touch event's action is 0x1 (deviceType=0) [pCnt=1, s=] when=1882020620369000
12-06 12:07:08.997 2795-2855/? I/InputDispatcher: Delivering touch to (26417): action: 0x1, toolType: 1
12-06 12:07:09.017 26417-26417/? D/com.distriqt.Firebase.Crash: CrashController::log( Testing an ANE log message. TEST TEST TEST TEST TEST )
                                                                
                                                                --------- beginning of system
12-06 12:07:09.357 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.357 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.357 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.357 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.357 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.357 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.377 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.377 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.377 2795-4370/? D/SSRM:n: SIOP:: AP = 450, LCD = 199
12-06 12:07:09.377 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.377 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.377 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.377 2795-3333/? E/WindowState: getStack: Window{2bbb9e66 u0 d0 air.com.laaser.psapliv/air.com.laaser.psapliv.AppEntry} couldn't find taskId=332 Callers=com.android.server.wm.WindowState.getDisplayContent:1066 com.android.server.wm.WindowState.isDefaultDisplay:1791 com.android.server.wm.WindowState.getMultiWindowStyleLw:2403 com.android.internal.policy.impl.multiwindow.MultiPhoneWindowManager.shouldEnableLayoutInsetsBySoftInput:1514 
12-06 12:07:09.387 2939-3449/? D/mali_winsys: new_window_surface returns 0x3000,  [1920x164]-format:1
12-06 12:07:09.397 2939-3449/? D/OpenGLRenderer: SRIB_DCS OpenGLRenderer getDcsFlag flag=0
12-06 12:07:09.407 2157-2157/? D/libEGL: eglInitialize EGLDisplay = 0xffed5f34
12-06 12:07:09.407 2157-2157/? D/libEGL: eglTerminate EGLDisplay = 0xffed5f94
12-06 12:07:09.677 2157-13548/? I/display: [PrimaryDisplay] [DYNAMIC_RECOMP] HWC_2_GLES by low FPS(3)
12-06 12:07:09.707 2795-2854/? V/AlarmManager: waitForAlarm result :8
12-06 12:07:09.707 2795-2820/? V/BroadcastQueue: [background] Process cur broadcast BroadcastRecord{1083b0ff u0 null} DELIVERED for app ProcessRecord{2df4496f 13932:com.google.android.gms.persistent/u0a9}
12-06 12:07:09.717 2795-3337/? D/ActivityManager: startService callerProcessName:com.google.android.gms, calleePkgName: com.google.android.gms
12-06 12:07:09.797 2795-3332/? D/ActivityManager: startService callerProcessName:com.google.android.gms, calleePkgName: com.google.android.gms
12-06 12:07:09.797 2157-2157/? I/display: [PrimaryDisplay] [DYNAMIC_RECOMP] GLES_2_HWC by high FPS(10)
12-06 12:07:09.817 13932-27029/? D/SecWifiDisplayUtil: Metadata value : SecSettings2

**Comment by marchbold** [comment]
SHA certificate fingerprints are found in your project settings. Select your android app and you'll see the fingerprints towards the bottom.

Yes you need to call `init` on all the extensions. If you don't do this none of the other functionality of the extension will work correctly.

- https://github.com/distriqt/ANE-Firebase/wiki/Core---Initialise

Similarly for any other firebase component you are using.

Try also enabling verbose analytics logging before posting your logs.

**Comment by jonathanhollander** [comment]
I see the SHA fingerprint thing your talking about, but I have no idea what to actually put because the only "certificate" i have for Android is a "p12" that flash builder created for me to sign the app. I have no idea how to get the P12 into that (00:00:00:00:00:00) format.

wow not sure how I messed needing to do the Firebase Core Initialize... very weird though that without doing that it WAS working for Android before on that original bundle ID, and still IS working for iOS. Either way I've now added it.

I am trying extremely hard to get the adb VERBOSE to work correctly but it just doesn't work when I do what you say. BTW, im on windows so I'm not sure if what you gave me was maybe for Mac. 

There are some extra steps your assuming I'm doing right? I'm listing EVERYthing im doing here:

Step1- Run the app in debug mode in Flash Builder (land on my default app main screen).

Step2- run command prompt 

Step3 - Type "adb shell setprop debug.firebase.analytics.app air.com.laaser.psapliv"
( I see nothing happen)

Step4 - Type "adb shell setprop log.tag.FA VERBOSE"
(I see nothing happen)

Step5 - Kill the app running in flash builder, and rerun the app in flash builder.

Step6 -  When I type "adb logcat"... the screen rapidly flies by with what seems like thousands and thousands of lines of code... but when I scroll up in the command prompt I only see the last say 500 lines.

Step7- (i found this online) instead type "adb logcat", type "adb logcat -d > logcat.txt"

This seems to dump everything into a file, instead of just the last 500 or so lines.

BUUTTTT. it seems like the whole "VERBOSE" thing didn't work, as I see everything in the log. I saw online that you could do "adb logcat -c" and it clears it. I then redid it and it seems to be a much smaller file, but it's still filled with everything. Attached is the "redone" logcat.
[logcat5.txt](https://github.com/distriqt/ANE-Firebase/files/1536959/logcat5.txt)

**Comment by jonathanhollander** [comment]
I found online that you can use Android Studio to try to do the VERBOSE thing and it sorta works I think.... This is what I see. Don't see any errors though

12-06 16:45:28.087 3829-4527/? V/FA: Recording user engagement, ms: 724865
12-06 16:45:28.087 3829-4527/? V/FA: Using measurement service
12-06 16:45:28.087 3829-4527/? V/FA: Connecting to remote service
12-06 16:45:28.097 3829-4527/? V/FA: Activity paused, time: 18329651493
12-06 16:45:28.097 3829-4527/? D/FA: Logging event (FE): user_engagement(_e), Bundle[{firebase_event_origin(_o)=auto, engagement_time_msec(_et)=724865, firebase_screen_class(_sc)=AppEntry, firebase_screen_id(_si)=-4732724145972149379}]
12-06 16:45:28.117 3829-4527/? V/FA: Using measurement service
12-06 16:45:28.117 3829-4527/? V/FA: Connection attempt already in progress
12-06 16:45:28.147 3829-4527/? D/FA: Connected to remote service
12-06 16:45:28.147 3829-4527/? V/FA: Processing queued up service tasks: 2
12-06 16:46:05.417 4941-4941/? I/FA: App measurement is starting up, version: 11020
12-06 16:46:05.417 4941-4941/? I/FA: To enable debug logging run: adb shell setprop log.tag.FA VERBOSE
12-06 16:46:05.447 4941-4941/? V/FA: Collection enabled
12-06 16:46:05.447 4941-4941/? V/FA: App package, google app id: air.com.laaser.psapliv, 1:804944061004:android:5c486800d686b1c6
12-06 16:46:05.447 4941-4941/? I/FA: Faster debug mode event logging enabled. To disable, run:
                                       adb shell setprop debug.firebase.analytics.app .none.
12-06 16:46:05.447 4941-4941/? D/FA: Debug-level message logging enabled
12-06 16:46:05.467 4941-4941/? V/FA: Registered activity lifecycle callback
12-06 16:46:05.477 4941-4997/? V/FA: Using measurement service
12-06 16:46:05.477 4941-4997/? V/FA: Connecting to remote service
12-06 16:46:05.507 4941-4997/? V/FA: Using measurement service
12-06 16:46:05.507 4941-4997/? V/FA: Connection attempt already in progress
12-06 16:46:05.507 4941-4941/? V/FA: onActivityCreated
12-06 16:46:05.687 4941-4997/? V/FA: Using measurement service
12-06 16:46:05.687 4941-4997/? V/FA: Connection attempt already in progress
12-06 16:46:05.687 4941-4997/? V/FA: Activity resumed, time: 18329689095
12-06 16:46:05.697 4941-4997/? I/FA: Tag Manager is not found and thus will not be used
12-06 16:46:05.697 4941-4997/? D/FA: Logging event (FE): screen_view(_vs), Bundle[{firebase_event_origin(_o)=auto, firebase_screen_class(_sc)=AppEntry, firebase_screen_id(_si)=-3528751347917879124}]
12-06 16:46:05.707 4941-4997/? V/FA: Using measurement service
12-06 16:46:05.707 4941-4997/? V/FA: Connection attempt already in progress
12-06 16:46:07.617 4941-4997/? D/FA: Connected to remote service
12-06 16:46:07.617 4941-4997/? V/FA: Processing queued up service tasks: 4
12-06 16:46:12.787 4941-4997/? V/FA: Inactivity, disconnecting from the service

**Comment by marchbold** [comment]
So see this line:

```
12-06 16:46:05.447 4941-4941/? V/FA: App package, google app id: air.com.laaser.psapliv, 1:804944061004:android:5c486800d686b1c6
```

That doesn't match the id in your google-services json which is `1:804944061004:android:b88f407d0551f4c7`

That would mean something is wrong in your config resources. Check the values in your custom resources ane match the values in the google services json.

**Comment by jonathanhollander** [comment]
wow you are 100% right.  And this perfectly explains why it worked for the other bundle id and not the new 1...when copying over the new values I only looked at the beginning part of that string and saw it the same as the other and assumed the rest was right (thinking it was some universal value)... It's amazing because I've gone line by line so many times and still never saw this. 

Like always thank you for your patience/support/kindness in dealing with mistakes your users make, you should know whenever I can I recommend you guys in any ANE disccussion. Keep setting the bar so high and I wont be able to do pullups anymore :)

**Comment by marchbold** [comment]
Nice! Glad I could help!

**Comment by jonathanhollander** [comment]
Have you ever seen this actually take the full 24 hours to show up? I thought I fixed this about 16-17 hours ago because I verified the debug events are correctly coming through in Firebase but I still see the "Your analytics data will appear here soon"... my higher ups don't believe Firebase when it says 24 hours and still think something is wrong (based off their own feelings nothing concrete), do you think I just still need to wait?

**Comment by marchbold** [comment]
Not to my knowledge, if you are getting the debug events we have never not seen them come through to the dashboard.

I have seen it take what I thought was over a day for data to appear in the dashboard though. Firebase seems really slow with that aspect of it's service and we have actually used Google Analytics as well because of this.

**Comment by jonathanhollander** [comment]
ok thanks again, will wait!
